src/algorithms/signal_source/gnuradio_blocks/fifo_reader.h

```diff
@@ -93,6 +67,9 @@ class FifoReader : virtual public gr::sync_block
     const std::string file_name_;
     const std::string sample_type_;
     std::ifstream fifo_;
+    char *fifo_buffer_;
```

Comment:
don't use naked pointers. If you want dynamic sizing, use `std::vector<char>`. If you are ok with static sizing, use `std::array<char>`. Or, if you must, use `std::unique_ptr<char[]>` (see [SO](https://softwareengineering.stackexchange.com/questions/380176/should-i-use-a-unique-ptr-with-an-array-type-or-a-vector))

Reply:
or maybe you want a `boost::circular_buffer`?

Reply:
Not applicable anymore after [77c682e](https://github.com/gnss-sdr/gnss-sdr/pull/584/commits/77c682e8b312c645d77e60289b0d67b33e67c633).